### PR TITLE
YARN-11313. [Federation] Add SQLServer Script and Supported DB Version in Federation.md.

### PR DESCRIPTION
**Pull request by slfan1989**
JIRA: YARN-11313. [Federation] Add SQLServer Script and Supported DB Version in Federation.md.

The SQLFederationStateStore in Yarn Federation supports Mysql and SqlServer, but the script provided by SqlServer lacks 6 scripts compared to Mysql's script.

1.dropDatabase.sql
2.dropStoreProcedures.sql
3.dropTables.sql
4.dropUser.sql
5.FederationStateStoreDatabase.sql
6.FederationStateStoreUser.sql

Supported database versions are not specified in Federation.md.

### CODE REVIEW

**Comment by slfan1989** [comment]
I will provide a test report of the newly added script as soon as possible.

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site/src/site/markdown/Federation.md [hunk 1]
```diff
@@ -205,9 +205,25 @@ In the same directory we provide scripts to drop the Stored Procedures, the Tabl
 
 **Note:** the FederationStateStoreUser.sql defines a default user/password for the DB that you are **highly encouraged** to set this to a proper strong password.
 
+The versions supported by MySQL are as follows:
+
+1. MySQL 5.7
+2. MySQL 8.0
+MySQL 5.7 and above can be used.
```
Comment (goiri) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site/src/site/markdown/Federation.md]: This text and the note are a little redundant.

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site/src/site/markdown/Federation.md [hunk 2]
```diff
@@ -205,9 +205,25 @@ In the same directory we provide scripts to drop the Stored Procedures, the Tabl
...
 SQL-Server scripts are located in **sbin/FederationStateStore/SQLServer/**.
 
+The versions supported by SQL-Server are as follows:
+
+1. SQL Server 2008 R2 Enterprise
+2. SQL Server 2012 Enterprise
+3. SQL Server 2016 Enterprise
+4. SQL Server 2017 Enterprise
+5. SQL Server 2019 Enterprise
+SQL Server 2008 R2 and above can be used.
```
Comment (goiri) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site/src/site/markdown/Federation.md]: Isn't this implied by the given list?

File: hadoop-yarn-project/hadoop-yarn/bin/FederationStateStore/SQLServer/FederationStateStoreDatabase.sql [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+-- Script to create a new Database in SQLServer for the Federation StateStore
+
+IF DB_ID ( '[FederationStateStore]') IS NOT NULL
```
Comment (goiri) [inline hadoop-yarn-project/hadoop-yarn/bin/FederationStateStore/SQLServer/FederationStateStoreDatabase.sql:21]: Are there some tests that now work?

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site/src/site/markdown/Federation.md (hunk 1 above)
Comment (slfan1989) [reply hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site/src/site/markdown/Federation.md]: Thanks for your help reviewing the code, I will modify this part.

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site/src/site/markdown/Federation.md (hunk 2 above)
Comment (slfan1989) [reply hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site/src/site/markdown/Federation.md]: I will provide this part as a list.

File: hadoop-yarn-project/hadoop-yarn/bin/FederationStateStore/SQLServer/FederationStateStoreDatabase.sql (hunk 3 above)
Comment (slfan1989) [reply hadoop-yarn-project/hadoop-yarn/bin/FederationStateStore/SQLServer/FederationStateStoreDatabase.sql:21]: I will execute the statement directly in the database of SqlServer and provide the test report.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 36s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  markdownlint  |   0m  0s |  |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  15m 33s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  25m 56s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   5m 36s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  66m 45s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 29s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   5m  8s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |   5m 16s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  27m 42s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 45s |  |  The patch does not generate ASF License warnings.  |
|  |   | 103m 29s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4927/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/4927 |
| Optional Tests | dupname asflicense codespell detsecrets mvnsite markdownlint |
| uname | Linux 99306aa246b0 4.15.0-191-generic #202-Ubuntu SMP Thu Aug 4 01:49:29 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / e93e92d93e6b9e563c0c8fe28e0052127c95e2eb |
| Max. process+thread count | 661 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site U: hadoop-yarn-project/hadoop-yarn |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4927/3/console |
| versions | git=2.25.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site/src/site/markdown/Federation.md (hunk 2 above)
Comment (goiri) [reply hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site/src/site/markdown/Federation.md]: I would order it the other way:
```
The versions supported by SQL-Server are SQL Server 2008 R2 and above:

1. SQL Server 2008 R2 Enterprise
2. SQL Server 2012 Enterprise
3. SQL Server 2016 Enterprise
4. SQL Server 2017 Enterprise
5. SQL Server 2019 Enterprise
```

Same for MySQL

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 40s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  markdownlint  |   0m  0s |  |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  15m 39s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  25m 54s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   5m 33s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  67m  4s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 33s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   5m  2s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |   5m 25s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  25m  0s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 46s |  |  The patch does not generate ASF License warnings.  |
|  |   | 100m 40s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4927/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/4927 |
| Optional Tests | dupname asflicense codespell detsecrets mvnsite markdownlint |
| uname | Linux 21c5d64228e5 4.15.0-191-generic #202-Ubuntu SMP Thu Aug 4 01:49:29 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / e93e92d93e6b9e563c0c8fe28e0052127c95e2eb |
| Max. process+thread count | 555 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site U: hadoop-yarn-project/hadoop-yarn |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4927/4/console |
| versions | git=2.25.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site/src/site/markdown/Federation.md (hunk 2 above)
Comment (slfan1989) [reply hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site/src/site/markdown/Federation.md]: Thank you very much for your suggestion!

The modified preview style is as follows:

image

**Review by goiri — APPROVED** [review]

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  14m  9s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  markdownlint  |   0m  0s |  |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  15m 47s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  28m  0s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   5m 56s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  71m 41s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 28s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   5m 26s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |   5m 56s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  29m 25s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 51s |  |  The patch does not generate ASF License warnings.  |
|  |   | 123m 37s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4927/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/4927 |
| Optional Tests | dupname asflicense codespell detsecrets mvnsite markdownlint |
| uname | Linux a19e983f5d6f 4.15.0-191-generic #202-Ubuntu SMP Thu Aug 4 01:49:29 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / ffe076dc31f3eb0c5afbeb49a7e9d13aef29ab8b |
| Max. process+thread count | 704 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site U: hadoop-yarn-project/hadoop-yarn |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4927/5/console |
| versions | git=2.25.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by slfan1989** [comment]
@goiri Thank you very much for your help reviewing the code!

**Pull request merged**